src/slurm_plugin/fleet_manager.py

```diff
@@ -0,0 +1,300 @@
+
+    def _evaluate_launch_params(self, count, launch_overrides):
+        """Evaluate parameters to be passed to create_fleet call."""
+        overrides = []
+        for instance_type in self._compute_resource_config.get("InstanceTypeList", []):
+            overrides.append({"InstanceType": instance_type["InstanceType"]})
+
+        launch_params = {
+            "LaunchTemplateConfigs": [
+                {
+                    "LaunchTemplateSpecification": {
+                        # LaunchTemplate is different for every compute resources in every queue
+                        "LaunchTemplateName": f"{self._cluster_name}-{self._queue}-{self._compute_resource}",
+                        "Version": "$Latest",
+                    },
+                    "Overrides": overrides,
+                }
+            ],
+            "TargetCapacitySpecification": {
+                "TotalTargetCapacity": count,
+                "DefaultTargetCapacityType": "on-demand"
+                if self._queue_config["CapacityType"] == "ONDEMAND"
+                else "spot",
+            },
+            "Type": "instant",
+            "SpotOptions": {
+                "AllocationStrategy": self._queue_config["AllocationStrategy"],
+                "SingleInstanceType": False,
+                "SingleAvailabilityZone": True,  # Set to False for Multi-AZ support
+                # If the minimum target capacity is not reached, the fleet launches no instances
+                "MinTargetCapacity": 1 if not self._all_or_nothing else count,
+            },
+            "OnDemandOptions": {
+                "AllocationStrategy": self._queue_config["AllocationStrategy"],
+                "CapacityReservationOptions": {"UsageStrategy": "use-capacity-reservations-first"},
+                "SingleInstanceType": False,
+                "SingleAvailabilityZone": True,  # Set to False for Multi-AZ support
+                # If the minimum target capacity is not reached, the fleet launches no instances
+                "MinTargetCapacity": 1 if not self._all_or_nothing else count,
+            },
+        }
+
+        launch_params.update(launch_overrides)
+        if launch_overrides:
+            logger.info("Found CreateFleet parameters override. Launching instances with: %s", launch_params)
+        return launch_params
+
+    def _launch_instances(self, launch_params):
+        """Launch a batch of ec2 instances."""
+        try:
+            response = create_fleet(self._region, self._boto3_config, launch_params)
+            for error in response.get("Errors", []):
+                logger.error("Error in CreateFleet request: %s - %s", error.get("ErrorCode"), error.get("ErrorMessage"))
+
+            logger.debug("CreateFleet response: %s", response)
+            instance_ids = [inst_id for instances in response["Instances"] for inst_id in instances["InstanceIds"]]
+            return {"Instances": self._get_instances_info(instance_ids)}
```

Comment:
in case we can't get the info for a single instance out of the entire batch it would be still ideal to avoid aborting the entire instance launch. Can we make it so that we return the instances we could describe and log an error for the missing ones that will be eventually terminated at the next iteration?

Reply:
Done, removed the raising of the Exception.